xbmc/URL.cpp

```diff
@@ -494,6 +488,16 @@ const std::string CURL::GetFileNameWithoutPath() const
   return URIUtils::GetFileName(file);
 }
 
+inline
+void protectIPv6(std::string &hn)
```

Comment:
@Memphiz 
yes, exactly as you said. I think we don't need more. This is what leaded me to this:
- we are working with already parsed and stored hostname.
- that means three scenarios. IPv4 IP, IPv6 IP or HostName.
- until we implemented IPv6 a colon was literally no-go
- because the 1), and previous one we know that :PORT is already stripped out and only case of having ":" on valid address is case of being IPv6
- then don't forget we do store the IP in it's canonised form.
- canonisation happens here https://github.com/xbianonpi/xbian-sources-xbmc/blob/treeHelix/xbmc/URL.cpp#L279
(this is part which will come after merging the networking code - while the canonise function checks the string to be formally validly stored IPv6 address)
- that means the stored hostname containing colons is IPv6 and according to RFC the canonised form must contain at minimum two colons.

(so even test for one colon would be formally fine from pure code perspective, but I thought I keep check for two as instant remainder we deal with proper text representation of IPv6 address)

are we fine ?